src/foam/u2/Element.js

```diff
@@ -625,6 +627,7 @@ foam.CLASS({
         An unloaded Element can be readded to the DOM.
       `,
       name: 'UNLOADED',
+      of: 'foam.u2.UnloadedElementState',
```

Comment:
Yeah me either, more often I think of would just be ElementState not the specific instance but in this case you need to caprue all the deps.

What context is the factory run in?  Would this.UnloadedState work if UnloadedState was required?

Furthermore I actually tried to model state machines like this in the past so it was much more high level and explicit rather than being constants and stuff, but I never managed to get it to a state that I felt it was good enough :)

Reply:
Good question. Currently, `this` is just the window so this.UnloadedState won't be a thing. Running these factories from a context that has these classes in them is something worth considering but not sure I really care to go down that road just yet. I'll need to get annoyed a little more first haha.